TrafficCapture/trafficCaptureProxyServer/src/main/java/org/opensearch/migrations/trafficcapture/proxyserver/netty/ExpiringSubstitutableItemPool.java

```diff
@@ -0,0 +1,304 @@
+package org.opensearch.migrations.trafficcapture.proxyserver.netty;
+
+import io.netty.channel.EventLoop;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.GenericFutureListener;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayDeque;
+import java.util.LinkedHashSet;
+import java.util.Queue;
+import java.util.StringJoiner;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+
+/**
+ * This class maintains N items for S seconds.  After S seconds, items are expired as per the
+ * specified expiration callback.  Callers can retrieve items from the cache or built on-demand
+ * if no items are available within the cache that are ready to go.
+ *
+ * This class does not use locking.  Instead, it is assumed that one of these will be created for
+ * each netty event loop.
+ */
+@Slf4j
+public class ExpiringSubstitutableItemPool<F extends Future<U>, U> {
+
+    private static class Entry<F> {
+        Instant timestamp;
+        F future;
+        public Entry(F future) {
+            timestamp = Instant.now();
+            this.future = future;
+        }
+
+        @Override
+        public String toString() {
+            final StringBuilder sb = new StringBuilder("Entry{");
+            sb.append("timestamp=").append(timestamp);
+            sb.append(", value=").append(future);
+            sb.append('}');
+            return sb.toString();
+        }
+    }
+
+    public static class PoolClosedException extends RuntimeException { }
+
+    public static class Stats {
+        @Getter
+        private long nItemsCreated;
+        @Getter
+        private long nExpiredItems;
```

Comment:
very minor nit, but this would match the previous item better if it were `nItemsExpired`

Reply:
fixed - thx